web/src/message_list_data.ts

```diff
@@ -46,6 +46,11 @@ export class MessageListData {
     // This is a callback that is called when messages are added to the message list.
     add_messages_callback?: (messages: Message[], recipient_order_changed: boolean) => void;
 
+    // ID of the MessageList for which this MessageListData is the data.
+    // Using ID instead of the MessageList object itself to avoid circular dependencies
+    // and to allow MessageList objects to be easily garbage collected.
+    rendered_message_list_id: number | undefined;
```

Comment:
Would it be cleaner to just clear this when a message list stops being rendered, rather than allowing this to be a stale integer ID for a message list that's been garbage-collected?

Reply:
Okay, updated the PR with that.